### PR TITLE
Add mqtt event example with value_template

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add mqtt event example with value_template

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #https://github.com/home-assistant/core/issues/98653

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/event.mqtt.markdown [hunk 1]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
...
+  - event:
+      name: "Desk button"
+      state_topic: "desk/button/state"
+      event_types:
+        - single
+        - double
+      device_class: "button"
+      value_template: |
+        { {% for key in value_json %}
+        {% if "Action" in value_json[key] %}
```
Comment (jbouwh) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
        {% if value_json[key].Action %}
```

File: source/_integrations/event.mqtt.markdown [hunk 2]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
 ```bash
 mosquitto_pub -h 127.0.0.1 -t home/doorbell/state -m '{"event_type": "press", "duration": 0.1}'
 ```
+
+### Example processing event data using a value template
```
Comment (c0ffeeca7) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
### Example: processing event data using a value template
```

File: source/_integrations/event.mqtt.markdown [hunk 3]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
...
+
+In many cases translation of an existing published payload is needed.
```
Comment (c0ffeeca7) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
In many cases, translation of an existing published payload is needed.
```

File: source/_integrations/event.mqtt.markdown [hunk 4]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
...
+The example config below translates payload `{"Button1": {"Action": "SINGLE"}}` of
```
Comment (c0ffeeca7) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
The example config below translates the payload `{"Button1": {"Action": "SINGLE"}}` of
```

File: source/_integrations/event.mqtt.markdown [hunk 5]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
...
+device `Button1` with event type `single` to the required format.
```
Comment (c0ffeeca7) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
the device `Button1` with event type `single` to the required format.
```

File: source/_integrations/event.mqtt.markdown [hunk 6]
```diff
@@ -221,3 +221,33 @@ The example below demonstrates how event attributes can be added to the event da
...
+An extra attribute `button` will be set to `Button1` and be added to the entity,
+but only if the `Action` property is set. Empty dict's will be ignored.
```
Comment (c0ffeeca7) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
but only if the `Action` property is set. Empty dictionaries will be ignored.
```

**Review by c0ffeeca7 — APPROVED** [review]
LGTM, @jbouwh :+1:

**Pull request merged**